sync/src/synchronizer/block_process.rs

```diff
@@ -32,14 +34,72 @@ impl<'a> BlockProcess<'a> {
         let shared = self.synchronizer.shared();
         let state = shared.state();
 
-        if state.new_block_received(&block) {
-            if let Err(err) = self.synchronizer.process_new_block(block.clone()) {
+        if !shared.active_chain().is_initial_block_download() {
+            // determine should we drop the queue and consumer handle
+
+            let mut queue_is_empty = false;
+            if let Some(queue) = self.synchronizer.block_queue.read().unwrap().as_ref() {
```

Comment:
We have specified that unwrap is not allowed to appear anywhere other than in tests,

> Using expect instead of unwrap and providing good error messages can convey your intent and make tracking down the source of a panic easier. ——https://doc.rust-lang.org/book/ch09-02-recoverable-errors-with-result.html

Reply:
more https://rust-lang.github.io/rust-clippy/master/index.html#unwrap_used